ocrd_segment/repair.py

```diff
@@ -89,9 +89,20 @@ def process(self):
 
             if plausibilize:
                 new_regions = []
+                reading_order = {}
+                # the reading order does not have to include all regions
+                # but it may include all types of regions!
+                for elem in page.get_ReadingOrder().get_OrderedGroup().get_RegionRefIndexed():
+                    reading_order[elem.get_regionRef()] = elem
                 for i in range(0,len(regions)):
                     if not i in mark_for_deletion:
                         new_regions.append(regions[i])
+                    else:
+                        if regions[i].get_id() in reading_order:
+                            page.get_ReadingOrder().get_OrderedGroup().get_RegionRefIndexed().remove(reading_order[regions[i].get_id()])
+                page.get_ReadingOrder().get_OrderedGroup().get_RegionRefIndexed().sort(key=RegionRefIndexedType.get_index)
+                for i in range(0, len(page.get_ReadingOrder().get_OrderedGroup().get_RegionRefIndexed())):
```

Comment:
Is it really necessary to correct the index if we are just removing? That is, does `RegionRefIndexed/@index` have to be contiguous? PAGE-XML is not specific here, but it may be implied and used by implementations:
> <documentation>Position (order number) of this item within the current hierarchy level.</documentation>

Reply:
Okay, I guess it must/should.